austation/code/modules/mining/minebot_upgrades.dm

```diff
@@ -0,0 +1,269 @@
+	incompatible with:\
+	Plasma tool conversion\
+	Barometric Field Generator"
+	upgrade_id = MBU_GUN_A
+	incompatible = list(MBU_GUN_B)
+
+/obj/item/minebot_upgrade/gun_a/upgrade_bot(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	. = ..()
+	if(!istype(M.minebot_gun, /obj/item/gun/energy/kinetic_accelerator))
+		return
+	var/obj/item/gun/energy/kinetic_accelerator/K = M.minebot_gun
+	K.overheat_time = 5
+
+/obj/item/minebot_upgrade/gun_a/uninstall(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	var/obj/item/gun/energy/kinetic_accelerator/K = M.minebot_gun
+	K.overheat_time = 10
+
+/obj/item/minebot_upgrade/gun_b
+	name = "Barometric Field Generator"
+	desc = "pressurises the PKA for a meatier attack; damage is increased to triplicate.\
+	incompatible with:\
+	Plasma tool conversion\
+	Barometric Field Generator"
+	upgrade_id = MBU_GUN_B
+	incompatible = list(MBU_GUN_A)
+
+/obj/item/minebot_upgrade/gun_b/upgrade_bot(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	. = ..()
+	if(!istype(M.minebot_gun, /obj/item/gun/energy/kinetic_accelerator))
+		return
+	var/obj/item/gun/energy/kinetic_accelerator/K = M.minebot_gun
+	K.ammo_type = /obj/item/ammo_casing/kinetic/heavy
+
+/obj/item/minebot_upgrade/gun_b/uninstall(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	if(!istype(M.minebot_gun, /obj/item/gun/energy/kinetic_accelerator))
+		return
+	var/obj/item/gun/energy/kinetic_accelerator/K = M.minebot_gun
+	K.ammo_type = /obj/item/ammo_casing/kinetic
+
+/obj/item/minebot_upgrade/gun_c
+	name = "Plasma tool conversion"
+	desc = "plasmacutters are the real miner's choice of tool; replaces that shoddy PKA.\
+	incompatible with:\
+	Auto-LODEr\
+	Barometric Field Generator"
+	upgrade_id = MBU_GUN_C
+	incompatible = list(MBU_GUN_A, MBU_GUN_B)
+
+/obj/item/minebot_upgrade/gun_c/upgrade_bot(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	. = ..()
+	if(istype(M.minebot_gun, /obj/item/gun/energy/plasmacutter))
+		to_chat(user, "<span class='notice'>the [M] already has a plasma gun.</span>")
+		return
+	if(uninstall(M, user))
+		var/obj/item/gun/energy/plasmacutter/P = new /obj/item/gun/energy/plasmacutter(src)
+		M.minebot_gun = P
+		P.cell = M.cell
+	else
+		to_chat(user, "<span class='notice'the previous weapon was not uninstalled.</span>")
+
+/obj/item/minebot_upgrade/gun_d
+	name = "Kinetic tool conversion"
+	desc = "reverts the minebot's armament to its original PKA."
+	upgrade_id = MBU_GUN_D
+
+/obj/item/minebot_upgrade/gun_d/upgrade_bot(mob/living/simple_animal/hostile/mining_drone/M, mob/user)
+	. = ..()
+	if(istype(M.minebot_gun, /obj/item/gun/energy/kinetic_accelerator))
+		to_chat(user, "<span class='notice'>the [M] already has a kinetic gun.</span>")
+		return
+	if(uninstall(M, user))
+		M.minebot_gun = new /obj/item/gun/energy/kinetic_accelerator(src)
+	else
+		to_chat(user, "<span class='notice'the previous weapon was not uninstalled.</span>")
```

Comment:
```suggestion
		to_chat(user, "<span class='notice'>The previous weapon was not uninstalled.</span>")
```

Reply:
Weird that I didn't commit this, but I went through both minebot.dm and minebot_upgrades.dm and gave both of  them the works.
This should be fixed now.